File: mac/VoiceTests/EnergyGateTests.swift
import XCTest
@testable import Voice

final class EnergyGateTests: XCTestCase {
    func testSilenceIsNotSpeech() {
        XCTAssertFalse(EnergyGate.hasSpeech([Float](repeating: 0, count: 16000)))
    }
    func testToneIsSpeech() {
        let tone = (0..<16000).map { Float(sin(Double($0) * 2 * .pi * 220 / 16000)) * 0.2 }
        XCTAssertTrue(EnergyGate.hasSpeech(tone))
    }
    func testOneLoudClickInSilenceIsNotSpeech() {
        var xs = [Float](repeating: 0, count: 16000)
        xs[8000] = 0.9
        XCTAssertFalse(EnergyGate.hasSpeech(xs)) // 95th percentile of 20 ms frames stays ~0
    }
}

File: windows/Spit.Core.Tests/EnergyGateTests.cs
namespace Spit.Core.Tests;

// Port of mac/VoiceTests/EnergyGateTests.swift.
public sealed class EnergyGateTests
{
    [Fact]
    public void testSilenceIsNotSpeech()
    {
        Assert.False(EnergyGate.HasSpeech(new float[16000]));
    }

    [Fact]
    public void testToneIsSpeech()
    {
        var tone = Enumerable.Range(0, 16000).Select(i => (float)Math.Sin(i * 2 * Math.PI * 220 / 16000) * 0.2f).ToArray();
        Assert.True(EnergyGate.HasSpeech(tone));
    }

    [Fact]
    public void testOneLoudClickInSilenceIsNotSpeech()
    {
        var xs = new float[16000];
        xs[8000] = 0.9f;
        Assert.False(EnergyGate.HasSpeech(xs)); // 95th percentile of 20 ms frames stays ~0
    }
}
